frontend/src/components/FilePreview/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { FilePreview as AutoFilePreview } from '@autonomys/auto-design-system';
+import {
+  OffchainMetadata,
+  canDisplayDirectly,
+  needsContentParsing,
+  processFileData,
+  decryptFileData,
+} from '@autonomys/auto-dag-data';
+import { useNetwork } from '../../contexts/network';
+import { NetworkId as AutoDriveNetworkId } from '../../constants/networks';
+import { NetworkId as AutoUtilsNetworkId } from '@autonomys/auto-utils';
+import { EXTERNAL_ROUTES } from '../../constants/routes';
+
+export const FilePreview = ({ metadata }: { metadata: OffchainMetadata }) => {
+  const { network } = useNetwork();
+  const [isFilePreview, setIsFilePreview] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [isDecrypted, setIsDecrypted] = useState(false);
+  const [decryptionError, setDecryptionError] = useState<string | null>(null);
+  const [file, setFile] = useState<Blob | null>(null);
+  const [textContent, setTextContent] = useState<string | null>(null);
+
+  const networkId = useMemo(() => {
+    switch (network.id) {
+      case AutoDriveNetworkId.LOCAL:
+      case AutoDriveNetworkId.MAINNET:
+        return AutoUtilsNetworkId.MAINNET;
+      case AutoDriveNetworkId.TAURUS:
+        return AutoUtilsNetworkId.TAURUS;
+      default:
+        return AutoUtilsNetworkId.MAINNET;
+    }
+  }, [network]);
+
+  const gatewayUrl = EXTERNAL_ROUTES.gatewayObjectDownload(metadata.dataCid);
+
+  const fetchFile = useCallback(
+    async (password?: string) => {
+      // If file is encrypted and no password provided, don't fetch
+      if (metadata.uploadOptions?.encryption && !password && !isDecrypted) {
+        setIsFilePreview(false);
+        setLoading(false);
+        return;
+      }
+
+      // For non-encrypted files that can be displayed directly,
+      if (!metadata.uploadOptions?.encryption && canDisplayDirectly(metadata)) {
+        setIsFilePreview(true);
+        setFile(null);
+        setLoading(false);
+        return;
+      }
+
+      // Encrypted files always need to be fetched and decrypted
+      setIsFilePreview(true);
+      setLoading(true);
+
+      try {
+        const response = await fetch(gatewayUrl);
```

Comment:
How are we handling very large file requests?

Reply:
We're not handling in a particular way. We could implement this mechanism for example:

Having two limits, the lower (e.g 1MB) is the maximum that could fetched from the DSN and the higher (e.g 100MB) would be biggest pre-viewable file.

For files <1MB would be pre-viewed always. For files between first and second limit will be previewed only if they're cached, and files beyond the 100MB limit the file won't be previewed.